flowmachine/tests/test_meaningful_locations.py

```diff
@@ -0,0 +1,299 @@
+    """ Test that column_names property matches head(0) for meaningfullocations"""
+    mfl = MeaningfulLocations(
+        clusters=HartiganCluster(
+            calldays=CallDays(
+                subscriber_locations=subscriber_locations(
+                    start="2016-01-01", stop="2016-01-02", level="versioned-site"
+                )
+            ),
+            radius=1,
+        ),
+        scores=EventScore(
+            start="2016-01-01", stop="2016-01-02", level="versioned-site"
+        ),
+        labels=labels,
+        label="evening",
+    )
+
+    assert get_column_names_from_run(mfl) == mfl.column_names
+
+
+def test_column_names_meaningful_locations_aggregate(
+    exemplar_level_param, get_column_names_from_run
+):
+    """ Test that column_names property matches head(0) for aggregated meaningful locations"""
+    if exemplar_level_param["level"] not in MeaningfulLocationsAggregate.allowed_levels:
+        pytest.xfail(
+            f'The level "{exemplar_level_param["level"]}" is not supported as an aggregation unit for MeaningfulLocations.'
+        )
+    mfl_agg = MeaningfulLocationsAggregate(
+        meaningful_locations=MeaningfulLocations(
+            clusters=HartiganCluster(
+                calldays=CallDays(
+                    subscriber_locations=subscriber_locations(
+                        start="2016-01-01", stop="2016-01-02", level="versioned-site"
+                    )
+                ),
+                radius=1,
+            ),
+            scores=EventScore(
+                start="2016-01-01", stop="2016-01-02", level="versioned-site"
+            ),
+            labels=labels,
+            label="evening",
+        ),
+        **exemplar_level_param,
+    )
+
+    assert get_column_names_from_run(mfl_agg) == mfl_agg.column_names
+
+
+def test_meaningful_locations_aggregate_disallowed_level_raises():
+    """ Test that a bad level raises a BadLevelError"""
+
+    with pytest.raises(BadLevelError):
+        mfl_agg = MeaningfulLocationsAggregate(
+            meaningful_locations=MeaningfulLocations(
+                clusters=HartiganCluster(
+                    calldays=CallDays(
+                        subscriber_locations=subscriber_locations(
+                            start="2016-01-01",
+                            stop="2016-01-02",
+                            level="versioned-site",
+                        )
+                    ),
+                    radius=1,
+                ),
+                scores=EventScore(
+                    start="2016-01-01", stop="2016-01-02", level="versioned-site"
+                ),
+                labels=labels,
+                label="evening",
+            ),
+            level="NOT_A_LEVEL",
+        )
+
+
+def test_column_names_meaningful_locations_od(
+    exemplar_level_param, get_column_names_from_run
+):
+    """ Test that column_names property matches head(0) for an od matrix between meaningful locations"""
```

Comment:
Again just a note to self: we have a bunch of tests that test this property for each query. We can probably encapsulate this better and ensure it is automatically valid?